### ISSUE TITLE
Generate Octomap from Gazebo world, how to install the needed plugin? 

### ISSUE THREAD

**Issue by andno037** [heronsystems/OpenMACE#12]
at https://github.com/heronsystems/OpenMACE/wiki/Generate-Octomap-from-Gazebo-world
I do not understand how to get the appropriate so file (libBuildOctomapPlugin.so).

**Comment by kkroeger007** [comment]
This is a plugin that is constructed when building the catkin_sim_environment workspace. There are also issues with between the latest gazebo/16.04/18.04 development environments that we are currently working through on a different branch. What dev environment are you using?

**Comment by andno037** [comment]
Thanks for respons. I'm working on with 18.04.
I actually manage to figure out that the libBuildOctomapPlugin.so is build by the sim_gazebo_pulgins. By some ugly copy and past ( Eigen path outside workspace :( ), I was able to build the application but it seems that may fast hack did not work. When I launch the gazebo with a world containing a <plugin = gazebo_octomap ...> tag I suspect that the output should indicate that the appropriate service is avalable right? As with the physics service bellow.

~/.gazebo/models/granso$ rosrun gazebo_ros gazebo granso.world 
[ INFO] [1580319624.146664635]: Finished loading Gazebo ROS API Plugin.
[ INFO] [1580319624.148662061]: waitForService: Service [/gazebo/set_physics_properties] has not been advertised, waiting...
[ INFO] [1580319628.574462260, 0.030000000]: waitForService: Service [/gazebo/set_physics_properties] is now available.
[ INFO] [1580319628.606734237, 0.060000000]: Physics dynamic reconfigure ready.

**Comment by kkroeger007** [comment]
@andno037 did you figure this out?

**Comment by andno037** [comment]
Yes, I was able to figure this out. needed move the so file to a location that gazebo was able to find it.

**Comment by kkroeger007** [comment]
Excellent. So were you just leveraging this project for that or is MACE providing something larger to you? Just curious.

**Comment by andno037** [comment]
Yes, I'm just needed to generate an octomap of the ground through. In order to comper with the collected octomap :).

**Comment by Malintha** [comment]
> Thanks for respons. I'm working on with 18.04.
> I actually manage to figure out that the libBuildOctomapPlugin.so is build by the sim_gazebo_pulgins. By some ugly copy and past ( Eigen path outside workspace :( ), I was able to build the application but it seems that may fast hack did not work. When I launch the gazebo with a world containing a <plugin = gazebo_octomap ...> tag I suspect that the output should indicate that the appropriate service is avalable right? As with the physics service bellow.
> 
> ~/.gazebo/models/granso$ rosrun gazebo_ros gazebo granso.world
> [ INFO] [1580319624.146664635]: Finished loading Gazebo ROS API Plugin.
> [ INFO] [1580319624.148662061]: waitForService: Service [/gazebo/set_physics_properties] has not been advertised, waiting...
> [ INFO] [1580319628.574462260, 0.030000000]: waitForService: Service [/gazebo/set_physics_properties] is now available.
> [ INFO] [1580319628.606734237, 0.060000000]: Physics dynamic reconfigure ready.

I am trying to get this one built on 18.04, Gazebo 7 environment. Seems like there are many changes from gazebo/math to ignition/math. I just wonder if there is a way around for that.

**Comment by malintha** [comment]
> This is a plugin that is constructed when building the catkin_sim_environment workspace. There are also issues with between the latest gazebo/16.04/18.04 development environments that we are currently working through on a different branch. What dev environment are you using?

I am trying to get this one built on 18.04, Gazebo 7 environment. Seems like there are many changes from gazebo/math to ignition/math. I just wonder if there is a way around for that.

**Comment by malintha** [comment]
This is the error I get when building sim_gazebo_plugins.
```
/home/malintha/OpenMACE/catkin_sim_environment/src/sim_gazebo_plugins/src/plugin_build_octomap.cpp:26:10: fatal error: gazebo/math/Vector3.hh: No such file or directory
 #include <gazebo/math/Vector3.hh>
          ^~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
make[2]: *** [CMakeFiles/BuildOctomapPlugin.dir/src/plugin_build_octomap.cpp.o] Error 1
make[1]: *** [CMakeFiles/BuildOctomapPlugin.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
/home/malintha/OpenMACE/catkin_sim_environment/src/sim_gazebo_plugins/src/plugin_stop_physics.cpp: In member function ‘virtual void gazebo::StopPhysicsPlugin::Load(gazebo::physics::WorldPtr, sdf::ElementPtr)’:
/home/malintha/OpenMACE/catkin_sim_environment/src/sim_gazebo_plugins/src/plugin_stop_physics.cpp:20:13: error: ‘class gazebo::physics::World’ has no member named ‘GetEnablePhysicsEngine’; did you mean ‘SetPhysicsEnabled’?
  if(_world->GetEnablePhysicsEngine())
             ^~~~~~~~~~~~~~~~~~~~~~
             SetPhysicsEnabled
/home/malintha/OpenMACE/catkin_sim_environment/src/sim_gazebo_plugins/src/plugin_stop_physics.cpp:23:11: error: ‘class gazebo::physics::World’ has no member named ‘EnablePhysicsEngine’
   _world->EnablePhysicsEngine(false);
           ^~~~~~~~~~~~~~~~~~~
make[2]: *** [CMakeFiles/StopPhysicsPlugin.dir/src/plugin_stop_physics.cpp.o] Error 1
make[1]: *** [CMakeFiles/StopPhysicsPlugin.dir/all] Error 2
make: *** [all] Error 2

```

**Comment by robofoxy** [comment]
Hello all,

My configuration is Ubuntu 18.04, Gazebo 9.16 and CMake version 3.5.2.

For the problem with 'gazebo/math/Vector3.hh',

* Checkout branch 0.0.2 from OpenMACE.

Then, you'll encounter with a CMake error about UUID,

* Remove cmake_modules package from find_package(catkin REQUIRED COMPONENTS ...)

And, it compiles.

**Comment by mateoteverga** [comment]
Hello, I am running Ubuntu 18.04 and ROS melodic. I noticed that when I launch the world via "rosrun gazebo_ros gazebo house.world" (with the plugin  "<plugin name='gazebo_octomap' filename='libBuildOctomapPlugin.so'/>"  included before the last <world>) the world generated in Gazebo is empty. And when I execute the rosservice call command, error says service not available. Any ideas of what am I doing wrong?

**Comment by mateoteverga** [comment]
Hello, I am running Ubuntu 18.04 and ROS melodic. I noticed that when I launch the world via "rosrun gazebo_ros gazebo house.world" (with the plugin included before the last world) the world generated in Gazebo is empty. And when I execute the rosservice call command, error says service not available. Any ideas of what am I doing wrong? Thanks!!

**Comment by emilcw** [comment]
@mateoteverga I have the same problem, when I try to launch the rosservice it has not been instantiated by the gazebo plugin. I suspect this due to that the libBuildOctomapPlugin.so file is not building correctly when running catkin_make. Any ideas?
